zetaclient/bitcoin_client.go

```diff
@@ -282,6 +282,41 @@ func (ob *BitcoinChainClient) WatchInTx() {
 	}
 }
 
+func (ob *BitcoinChainClient) postBlockHeader(tip int64) error {
+	ob.logger.WatchInTx.Info().Msgf("postBlockHeader: tip %d", tip)
+	bn := tip
+	res, err := ob.zetaClient.GetBlockHeaderStateByChain(ob.chain.ChainId)
+	if err == nil && res.BlockHeaderState != nil && res.BlockHeaderState.EarliestHeight > 0 {
+		bn = res.BlockHeaderState.LatestHeight + 1
+	}
+	if bn > tip {
+		return fmt.Errorf("postBlockHeader: must post block confirmed block header: %d > %d", bn, tip)
```

Comment:
Does `bn > tip` mean that zetacore is waiting for block `bn` and zetaclient is lagging behind and doesn't need to do anything in here? I meant should we return an error or just return a `nil` to skip?

Reply:
the "tip" is supposed to be a few blocks behind what RPC returns the latest block number, essentially the "confirmed" block number. If bn > tip then we need to wait for block confirmation.